Library/Homebrew/dev-cmd/generate-cask-api.rb

```diff
@@ -44,22 +44,24 @@ def run
 
           Cask::Cask.generating_hash!
 
-          tap.cask_files.each do |path|
-            cask = Cask::CaskLoader.load(path)
-            name = cask.token
-            json = JSON.pretty_generate(cask.to_hash_with_variations)
-            cask_source = path.read
-            html_template_name = html_template(name)
+          Homebrew::SimulateSystem.with(os: MacOSVersion::SYMBOLS.keys.first, arch: :arm) do
```

Comment:
I wonder if what we actually want is the first version < `HOMEBREW_MACOS_NEWEST_UNSUPPORTED`. We'll be adding macOS 16 to the symbol list next month.

Reply:
It probably won't matter, because even if a new item is added to `SYMBOLS` in `macos_version.rb`, there won't be any macOS 16-specific changes in any formulae or casks until it's officially supported, so the output will be the same.

Reply:
There probably will be for some like `onyx` which last time we pointed to a beta in August: https://github.com/Homebrew/homebrew-cask/commit/3c0b4153e1d552078aad1901fd50db908a82231c. But indeed not too many.

Reply:
Slept on it, then implemented your suggestion.